Migrate GlobalAppConfig from couch to sql (3 of 3)

orangejenny opened the pull request:
Branched off of https://github.com/dimagi/commcare-hq/pull/26026, first new commit is "Removed couch view".

This removes the GlobalAppConfig couch view and related code.

orangejenny committed:
Removed couch view

millerdev reviewed (approved):
Approval conditional on https://github.com/dimagi/commcare-hq/pull/26025#discussion_r352876479 being resolved.

orangejenny committed:
Renamed SQLGlobalAppConfig

orangejenny committed:
Added override of table name during migration

orangejenny commented:
Rebased and added be584787c00e90cae45b1008b33179f5a605ae76 to ensure the correct table name during the django migration that enforces the data migration, similar to https://github.com/dimagi/commcare-hq/pull/26400/

Assuming tests pass, this should be good to merge and deploy tomorrow.

orangejenny commented:
Deployed to prod, verified each old couch doc has an equivalent SQL object.
